### PR TITLE
Cleaning code repetition in homebrew interface

### PR DESCRIPTION
**Pull request by xhruso00**
Breaking into reusable methods

### CODE REVIEW

File: Cakebrew/BPHomebrewInterface.m [hunk 1]
```diff
@@ -92,30 +89,101 @@ - (void)dealloc
...
 
+- (void)cleanup
+{
+  [self.task terminate];
+}
+
+- (BOOL)systemHasAppNap
+{
+  static BOOL systemHasAppNap;
+  if (!systemHasAppNap) {
```
Comment (brunophilipe) [inline Cakebrew/BPHomebrewInterface.m]: `systemHasAppNap` is not a pointer. If the system does not have AppNap, this if will be executed every time. You could replace it for an NSNumber though.

**Comment by xhruso00** [comment]
Thanks. Will fix this.

**Comment by xhruso00** [comment]
Fixed.

**Comment by brunophilipe** [comment]
Is it wise to let the user run multiple tasks simultaneously? This was originally not supported because it could leave brew in an inconsistent state. What do the home-brew devs say about it?

**Comment by brunophilipe** [comment]
Also please be careful as commits made in the same branch are appended to the pull request even after you opened it.

**Comment by xhruso00** [comment]
Read only is completely safe. Writing -> we still keep modal window. There is even project that does is concurrently. Besides that's why we have dev branch + git. There is even example
https://github.com/marcransome/MRBrew

This branch has some commits -> like SRI for NSTask with a nice delegate. Do I need to make another branch because I am bit confused. For the moment I stopped because it wouldn't make sense to continue without merge/approval/your consent

**Comment by xhruso00** [comment]
http://superuser.com/questions/428713/is-it-safe-to-run-multiple-brew-install-commands-at-the-same-time

**Pull request merged**